cms/djangoapps/contentstore/views/preview.py

```diff
@@ -291,6 +293,7 @@ def _studio_wrap_xblock(xblock, view, frag, context, display_name_only=False):
             'xblock': xblock,
             'show_preview': context.get('show_preview', True),
             'content': frag.content,
+            'is_library': is_library,
```

Comment:
Instead of adding this new context variable to the template, I would prefer that you correctly set `can_edit_visibility`, perhaps by changing the default on the line below:

```
'can_edit_visibility': context.get('can_edit_visibility', xblock.course_id.is_course),  # Default to true for courses, false for libraries
```
since it was pointed out on Jira that the move button has the same issue, we can fix it at the same time:

```
'can_move': context.get('can_move', xblock.course_id.is_course),
```

Reply:
I agree; the proposed solution was to avoid side effects based on context.